### PR TITLE
Added support of pending block identifier

### LINKED ISSUE #1578: Method `eth_getBlockByNumber` doesn't support `pending` parameter
Some projects in Ethereum uses parameter `pending` in `eth_getBlockByNumber` to get the transactions that are pending in mempool. Infura supports this parameter but RSK node doesn't:

```
curl -X POST -H "Content-Type:application/json" --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["pending", true],"id":1}' https://public-node.rsk.co
```

```
{"jsonrpc":"2.0","id":1,"error":{"code":-32201,"message":"This method doesn't support 'pending' as a parameter"}}
```

### PR DESCRIPTION
**Pull request by Vovchyk**
## Description

This PR adds support of the "pending" block identifier (see [The default block parameter](https://eth.wiki/json-rpc/API#the-default-block-parameter) section).

This capability has been introduced in the following JSON-RPC methods:

- `eth_getBlockByNumber`
- `eth_getTransactionByBlockNumberAndIndex`
- `eth_getUncleCountByBlockNumber`
- `eth_getUncleByBlockNumberAndIndex`

Fixes #1578.

## Motivation and Context

Besides already implemented "earliest" and "latest" block identifiers, add support of "pending", as described in the [spec](https://eth.wiki/json-rpc/API#the-default-block-parameter).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

File: rskj-core/src/main/java/co/rsk/rpc/Web3InformationRetriever.java [hunk 1]
```diff
@@ -41,35 +42,37 @@
...
      */
     public Optional<Block> getBlock(String identifier) {
-        if (PENDING.equals(identifier)) {
-            throw unimplemented("This method doesn't support 'pending' as a parameter");
-        }
-
         Block block;
-        if (LATEST.equals(identifier)) {
+        if (PENDING.equals(identifier)) {
+            block = executionBlockRetriever.getExecutionBlock_workaround(identifier).getBlock();
```
[the remark this answers is not in the document]
Comment (Vovchyk) [reply rskj-core/src/main/java/co/rsk/rpc/Web3InformationRetriever.java]: agree, but it seems that this method was deprecated by mistake at the time - it's was introduced as fix for another issue and right now is in use. I've removed the annotation out of the method. Do you see any issue with this method?

**Review by heitorfm — DISMISSED** [review]

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Review by heitorfm — APPROVED** [review]

**Pull request merged**